include/knowhere/index/index_node.h

```diff
@@ -62,9 +62,9 @@ class IndexNode : public Object {
      * to be revisited.
      */
     virtual Status
-    Build(const DataSetPtr dataset, const Config& cfg) {
+    Build(const DataSetPtr dataset, std::shared_ptr<Config> cfg) {
```

Comment:
Please add comments to all methods on why it is needed to pass `Config` this way. It is clear from the description of this PR, but it not very clear without it. 
Please also explain why you pick `std::shared_ptr<>` or `std::unique_ptr<>` depending on a use case.